hydrolib/core/utils.py

```diff
@@ -300,3 +301,34 @@ def _from_windows_to_posix_path(cls, windows_path: Path) -> str:
         posix_path = posix_root + "/".join(parts)
 
         return posix_path
+
+
+class FileChecksumCalculator:
+    """
+    FileChecksumCalculator calculator used to calculate the checksum of a file.
+    """
+
+    @staticmethod
+    def calculate_checksum(filepath: Path) -> Optional[str]:
+        """Calculate the checksum of the file from the given filepath.
+
+        Args:
+            filepath (Path): The filepath to the file for which the checksum will be calculated.
+
+        Returns:
+            [Optional[str]]:
+                The checksum of the file.
+                When the filepath doesn't exist or the filepath isn't a file, None.
+        """
+        if not filepath.exists() or not filepath.is_file():
+            return None
+
+        return FileChecksumCalculator._calculate_sha256_checksum(filepath)
+
+    @staticmethod
+    def _calculate_sha256_checksum(filepath: Path) -> str:
```

Comment:
Shouldn't a MD5 checksum be enough to check if there are differences? MD5 is faster and we do not need the cryptographic security SHA256 offers :p

Reply:
Discussed. We will change it to MD5.

Reply:
This has become a security risk in sonarcloud

Reply:
Marked it safe in sonarcloud, as it poses no security risk in our context.